apps/emqx_durable_storage/src/emqx_ds_optimistic_tx.erl

```diff
@@ -394,10 +394,9 @@ handle_event(info, {'EXIT', _, Reason}, _State, _Data) ->
         normal -> keep_state_and_data;
         _ -> {stop, shutdown}
     end;
-handle_event(enter, _OldState, ?leader(?pending), D0 = #d{flush_interval = T}) ->
-    %% Schedule unconditional flush after the given interval:
+handle_event(enter, _OldState, ?leader(?pending), D0) ->
     D = D0#d{entered_pending_at = erlang:monotonic_time(millisecond)},
-    {keep_state, D, {state_timeout, T, ?timeout_flush}};
```

Comment:
We only enter `?leader(?pending)` from `?leader(?idle)` setting early timeout explicitly